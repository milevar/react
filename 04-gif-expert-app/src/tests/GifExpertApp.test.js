import React from "react";
import {shallow} from "enzyme";
import {GifExpertApp} from "../GifExpertApp";


describe('Pruebas en <GifExpertApp />', () => {

    const category = 'Garfield';

    test('Debe de mostrar el componente correctamente', () => {

        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar lista de categorias', () => {

        const categories = ['Love', 'Garfield'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
})


