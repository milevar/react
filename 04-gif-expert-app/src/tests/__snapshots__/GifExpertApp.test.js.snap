// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe de mostrar lista de categor 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe de mostrar lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Love"
      key="Love"
    />
    <GifGrid
      category="Garfield"
      key="Garfield"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe de mostrar lista de categorias 2`] = `undefined`;
